### ISSUE TITLE
Simplifying CI config

### ISSUE THREAD

**Issue by piersy** [clearmatics/autonity#465]
# Problem

The github actions CI config is quite fragmented because there are many jobs which causes it to be very verbose and have a lot of duplication. This leads to hard to understand CI config, which in turn leads to mistakes. For example repeated tests #462 or missing tests #464.

The fragmentation also makes updating things that are common between jobs a lot of work. See https://github.com/clearmatics/autonity/pull/457/files#diff-25a57620242e0929b57dd916b7894b32 for the changes required to install a binary and generate contract source. This pull request shows how a simple change ends up as total overkill in the CI config. The result being that the CI as it stands can prevent us from achieving the desired processes.

# Implementation

The large number of jobs is due to the fact that we can run tests quicker by having many jobs which parallelise the work.

A possible quick fix is that there are a number of short testing jobs that could be combined and still execute in less time than the main test job.

Better than that though would be to tackle the source of the problem, which is slow tests. If we can speed up our tests sufficiently we should be able to put all tests into a single job, ideally we could get build and test into a single job.  

# Acceptance Criteria

Reduce the size of the ci yaml config.

# Notes
Probably speeding up tests deserves its own ticket and should be done first.

**Comment by yazzaoui** [comment]
Unfortunately, some tests scenarios require from us to have slow tests so I don't think we would be able to merge everything in one single job.
